From 184d7c79080c6291035b3169b24bbfcd76256e7d Mon Sep 17 00:00:00 2001
From: Neha Bisht <quic_nbisht@quicinc.com>
Date: Tue, 24 Jan 2023 13:21:24 +0530
Subject: [PATCH] skb_recycler: Enable reycler debug api under appropriate
 macro

Enable skb fast recycler debug api under reycler and debug check
macros

Change-Id: I0386ef6391761d2804f06b29b1d886b8ed0d117a
Signed-off-by: Neha Bisht <quic_nbisht@quicinc.com>
---
 include/linux/skbuff.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/linux/skbuff.h b/include/linux/skbuff.h
index c1cbdafd83f5..eb4ca84a3df2 100644
--- a/include/linux/skbuff.h
+++ b/include/linux/skbuff.h
@@ -1382,7 +1382,11 @@ static inline int skb_pad(struct sk_buff *skb, int pad)
 }
 #define dev_kfree_skb(a)	consume_skb(a)
 #define dev_kfree_skb_list_fast(a)	consume_skb_list_fast(a)
+#if defined(SKB_FAST_RECYCLABLE_DEBUG_ENABLE) && defined(CONFIG_SKB_RECYCLER)
 #define dev_check_skb_fast_recyclable(a)       check_skb_fast_recyclable(a)
+#else
+#define dev_check_skb_fast_recyclable(a)
+#endif
 
 int skb_append_pagefrags(struct sk_buff *skb, struct page *page,
 			 int offset, size_t size);
-- 
2.34.1

